src/gtk/choice.cpp

```diff
@@ -161,6 +163,8 @@ int wxChoice::DoInsertItems(const wxArrayStringsAdapter & items,
 
     int n = wxNOT_FOUND;
 
+    gtk_widget_freeze_child_notify(m_widget);
```

Comment:
Normally we prefer to use RAII helpers like `wxWindowUpdateLocker` for calling pairs of functions like this. Not sure if it's really needed if it's the only place where we're going to use it, but I think it might be useful elsewhere, in which case adding it to `include/wx/gtk/private.h` or some new file under `include/wx/gtk/private/` would be nice.

Reply:
It looks like this is the only use of the `gtk_widget_freeze_child_notify` and the equivalent thaw function, and I am not sure if there will be other uses of it (do you have any thoughts of other places to use it?). I'm tempted to say we leave it as a non-RAII wrapper here unless there are other places we want it right now.

Reply:
I thought it could be useful in `wxListBox::DoInsertItems()` too, but I didn't look at it in any details, so let's leave it like this until we need it there...